src/crimson/os/seastore/journal.cc

```diff
@@ -353,15 +353,15 @@ Journal::replay_ret Journal::replay(delta_handler_t &&delta_handler)
   return seastar::do_with(
     std::make_pair(std::move(delta_handler), std::vector<paddr_t>()),
     [this](auto &&item) mutable -> replay_ret {
-      auto &[handler, segments] = item;
       return find_replay_segments(
-      ).safe_then([this, &handler, &segments](auto osegments) {
+      ).safe_then([this, &item](auto osegments) {
+	auto &[handler, segments] = item;
 	logger().debug("replay: found {} segments", segments.size());
 	segments.swap(osegments);
 	return crimson::do_for_each(
 	  segments,
-	  [this, &handler](auto i) {
-	    return replay_segment(i, handler);
+	  [this, &item](auto i) {
+	    return replay_segment(i, item.first);
```

Comment:
instead of using labels, can we just capture handler and segments like
```c++
  return seastar::do_with(std::move(delta_handler), std::vector<paddr_t>(),
    [this](auto&& handler, auto&& segments) mutable -> replay_ret {
      return find_replay_segments().safe_then(
        [this, &handler, &segments](auto replay_segs) {
          logger().debug("replay: found {} segments", replay_segs.size());
          segments = std::move(replay_segs);
          return crimson::do_for_each(segments, [this, &handler](auto i) {
            return replay_segment(i, handler);
          });
        });
    });
```

i made 4 changes in the proposed code snippet:

1. capture `delta_handler` and segments in the outmost `do_with()` call, instead of holding them using a pair. so we don't need to unpack them using the structured binding to avoid the label/variable confusion. as per C++ standard, this binding creates a label instead of variables.
1. print the number of found segments, instead of the number of not-yet-changed segments. as i think, at that moment, `segments` is still empty
1. use `move()` instead `swap()` for assigning the value of `osegments` to `segments`, as we don't care about the status of `osegments` afterwards, so assigning the rvalue reference of `osegments` is easier to read.
1. i reformatted the code a little bit.

Reply:
Ok, adopted your version.